File: backend/lib/backend/benefits/order.ex
defmodule Backend.Benefits.Order do
  use Ecto.Schema
  import Ecto.Changeset

  schema "orders" do
    belongs_to :user, Backend.Benefits.User, type: :string, references: :user_id
    has_many :benefits, Backend.Benefits.Benefit
    field :total, :integer

    timestamps()
  end

  @doc false
  def changeset(order, attrs) do
    order
    |> cast(attrs, [:total])
    |> validate_required([:total])
  end
end

File: backend/app/models/order.rb
class Order < ApplicationRecord

  validates :total, presence: true, numericality: { greater_than_or_equal_to: 0 }
  validates :user_id, presence: true

  belongs_to :user
  has_many :purchaseds  
  has_many :products, through: :purchaseds
  
end
